summer_staj.py

```diff
@@ -74,24 +87,26 @@ def parse(self, response):
 
         stuff = []
 
-        size = len(soup.find_all("td", class_="companyName"))
+        all_rows = soup.find_all("tr", class_="company")
+        size = len(all_rows)
 
         former_id = 0  # this checks duplicates. thanks bilkent for the mistakes :(
 
-        for x in range(size):
-                company = soup.find_all("td", class_="companyName")[x]
-                companyTop = soup.find_all("tr", class_="company")[x]

```

Comment:
Same here, you can crawl the children of a single html tag instead of scanning the entire document again and again.